File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private void inorder(TreeNode root,int k,int[] count){
            if(root==null){
                return;
            }
            inorder(root.left,k,count);
            count[0]+=1;
            if(count[0]==k){
                 count[1]=root.val;
            }
            inorder(root.right,k,count);
        }
    public int kthSmallest(TreeNode root, int k) {
        int[] count=new int[]{0,-1};
         inorder(root,k,count);
         return count[1];
    }
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.res=[]
        def inOrder(root,k):
            if root is None:
                return
           
            inOrder(root.left,k)
            self.res.append(root.val)  
            inOrder(root.right,k)

        inOrder(root,k)
        return self.res[k-1]